blocksuite/affine/gfx/turbo-renderer/src/extension/index.ts

```diff
@@ -0,0 +1,19 @@
+import { createIdentifier } from '@blocksuite/global/di';
+import type { ExtensionType } from '@blocksuite/store';
+
+import type { BlockLayoutPainter } from '../types';
+
+export const BlockPainterProvider = createIdentifier<BlockLayoutPainter>(
+  'block-painter-provider'
+);
+
+export const BlockLayoutPainterExtension = (
+  type: string,
+  painter: new () => BlockLayoutPainter
```

Comment:
I tried my best to avoid nesting plain class impl inside extension. If we go this way, making `BlockLayoutPainter` itself `ExtensionType` could reduce boilerplate.